tests/integration_tests/clouds.py

```diff
@@ -165,19 +191,24 @@ def _perform_launch(self, launch_kwargs):
         pycloudlib_instance.wait(raise_on_cloudinit_failure=False)
         return pycloudlib_instance
 
-    def _mount_source(self, instance):
-        container_path = '/usr/lib/python3/dist-packages/cloudinit'
-        format_variables = {
-            'name': instance.name,
-            'cloudinit_path': cloudinit.__path__[0],
-            'container_path': container_path,
-        }
-        log.info(
-            'Mounting source {cloudinit_path} directly onto LXD container/vm '
-            'named {name} at {container_path}'.format(**format_variables))
-        command = (
-            'lxc config device add {name} host-cloud-init disk '
-            'source={cloudinit_path} '
-            'path={container_path}'
-        ).format(**format_variables)
-        subp(command.split())
+
+class LxdContainerCloud(_LxdIntegrationCloud):
+    datasource = 'lxd_container'
+
+    def _get_cloud_instance(self):
+        return LXDContainer(tag='lxd-container-integration-test')
```

Comment:
Just a nit, but we could make this even more declarative:

```suggestion
class LxdContainerCloud(_LxdIntegrationCloud):
    datasource = 'lxd_container'
    pycloudlib_instance_cls = LXDContainer
    instance_tag = "lxd-container-integration-test"
```

And add to the superclass:

```py
    def _get_cloud_instance(self):
        return self.pycloudlib_instance_cls(tag=self.instance_tag)
```